### PR TITLE
Deliverable B: Review for Coding conventions. GROUP D

### PR DESCRIPTION
**Pull request by revanth-sridhar**
Files assigned to Manoj are: 
- message.cpp (src/data_structures), 
- main.cpp (test/src/sender), 
- sendercadmium.hpp (include/atomics).

Files assigned to Revanth are: 
- receivercadmium.hpp (include/atomics),
- subnetcadmium.hpp (include/atomics),
- message.hpp (include/data_structures).

### CODE REVIEW

File: src/data_structures/message.cpp [hunk 1]
```diff
@@ -7,19 +7,19 @@
...
 /***************************************************/
-/************* Output stream ************************/
+/************* Output stream ***********************/
 /***************************************************/
 
-ostream& operator<<(ostream& os, const Message_t& msg) {
-  os << msg.value;
-  return os;
+ostream& operator<<(ostream& os, const message_t& msg) {
+    os << msg.value;
```
Comment (gagandeep8030) [inline src/data_structures/message.cpp:14]: name of the variable "os" could be "out_stream" to be more descriptive.

File: src/data_structures/message.cpp [hunk 2]
```diff
@@ -7,19 +7,19 @@
...
+    return os;
 }
 
 /***************************************************/
 /************* Input stream ************************/
 /***************************************************/
 
-istream& operator>> (istream& is, Message_t& msg) {
-  is >> msg.value;
```
Comment (gagandeep8030) [inline src/data_structures/message.cpp:23]: Similarly "is" could be replaced with "input_stream".The variable names should be self-explanatory.

File: src/top_model/main.cpp [hunk 3]
```diff
@@ -123,12 +123,12 @@ cadmium::dynamic::modeling::Ports iports_Network = {typeid(inp_1),typeid(inp_2)}
...
-  cadmium::dynamic::translate::make_EIC<inp_1, Subnet_defs::in>("subnet1"),
-  cadmium::dynamic::translate::make_EIC<inp_2, Subnet_defs::in>("subnet2")
+  cadmium::dynamic::translate::make_EIC<inp_1, subnet_defs::in>("subnet1"),
+  cadmium::dynamic::translate::make_EIC<inp_2, subnet_defs::in>("subnet2")
 };
 cadmium::dynamic::modeling::EOCs eocs_Network = {
-  cadmium::dynamic::translate::make_EOC<Subnet_defs::out,outp_1>("subnet1"),
-cadmium::dynamic::translate::make_EOC<Subnet_defs::out,outp_2>("subnet2")
+  cadmium::dynamic::translate::make_EOC<subnet_defs::out,outp_1>("subnet1"),
+cadmium::dynamic::translate::make_EOC<subnet_defs::out,outp_2>("subnet2")
```
Comment (gagandeep8030) [inline src/top_model/main.cpp:131]: Missing indentation of 4 spaces.

File: src/top_model/main.cpp [hunk 4]
```diff
@@ -146,19 +146,19 @@ std::shared_ptr<cadmium::dynamic::modeling::coupled<TIME>> NETWORK = std::make_s
...
+cadmium::dynamic::modeling::Models submodels_ABPSimulator = {sender1, receiver1, NETWORK};
 cadmium::dynamic::modeling::EICs eics_ABPSimulator = {
-  cadmium::dynamic::translate::make_EIC<inp_control, Sender_defs::controlIn>("sender1")
+  cadmium::dynamic::translate::make_EIC<inp_control, sender_defs::control_in>("sender1")
 };
 cadmium::dynamic::modeling::EOCs eocs_ABPSimulator = {
-  cadmium::dynamic::translate::make_EOC<Sender_defs::packetSentOut,outp_pack>("sender1"),
-cadmium::dynamic::translate::make_EOC<Sender_defs::ackReceivedOut,outp_ack>("sender1")
+  cadmium::dynamic::translate::make_EOC<sender_defs::packet_sent_out,outp_pack>("sender1"),
+cadmium::dynamic::translate::make_EOC<sender_defs::ack_received_out,outp_ack>("sender1")
```
Comment (gagandeep8030) [inline src/top_model/main.cpp:155]: Missing indentation of 4 spaces.

File: src/data_structures/message.cpp (hunk 1 above)
Comment (revanth-sridhar) [reply src/data_structures/message.cpp:14]: Many thanks for this feedback. It is resolved now.

File: src/data_structures/message.cpp (hunk 2 above)
Comment (revanth-sridhar) [reply src/data_structures/message.cpp:23]: Many thanks for this feedback. It is resolved now.

File: src/top_model/main.cpp [hunk 5]
```diff
@@ -123,12 +123,12 @@ cadmium::dynamic::modeling::Ports iports_Network = {typeid(inp_1),typeid(inp_2)}
...
+  cadmium::dynamic::translate::make_EIC<inp_2, subnet_defs::in>("subnet2")
 };
 cadmium::dynamic::modeling::EOCs eocs_Network = {
-  cadmium::dynamic::translate::make_EOC<Subnet_defs::out,outp_1>("subnet1"),
-cadmium::dynamic::translate::make_EOC<Subnet_defs::out,outp_2>("subnet2")
+  cadmium::dynamic::translate::make_EOC<subnet_defs::out,outp_1>("subnet1"),
+cadmium::dynamic::translate::make_EOC<subnet_defs::out,outp_2>("subnet2")
 };
 cadmium::dynamic::modeling::ICs ics_Network = {};
 std::shared_ptr<cadmium::dynamic::modeling::coupled<TIME>> NETWORK = std::make_shared<cadmium::dynamic::modeling::coupled<TIME>>(
```
Comment (gagandeep8030) [inline src/top_model/main.cpp:134]: Just want to make sure that this line has been arranged according to 80 maximum. characters. There are some more lines in these code too which could have more than 80 characters.

File: src/top_model/main.cpp [hunk 6]
```diff
@@ -123,12 +123,12 @@ cadmium::dynamic::modeling::Ports iports_Network = {typeid(inp_1),typeid(inp_2)}
 cadmium::dynamic::modeling::Ports oports_Network = {typeid(outp_1),typeid(outp_2)};
 cadmium::dynamic::modeling::Models submodels_Network = {subnet1, subnet2};
 cadmium::dynamic::modeling::EICs eics_Network = {
-  cadmium::dynamic::translate::make_EIC<inp_1, Subnet_defs::in>("subnet1"),
-  cadmium::dynamic::translate::make_EIC<inp_2, Subnet_defs::in>("subnet2")
+  cadmium::dynamic::translate::make_EIC<inp_1, subnet_defs::in>("subnet1"),
+  cadmium::dynamic::translate::make_EIC<inp_2, subnet_defs::in>("subnet2")
 };
 cadmium::dynamic::modeling::EOCs eocs_Network = {
-  cadmium::dynamic::translate::make_EOC<Subnet_defs::out,outp_1>("subnet1"),
```
Comment (gagandeep8030) [inline src/top_model/main.cpp:130]: The variables names like "outp_1"  could be more descriptive like "out_put1".

File: src/top_model/main.cpp [hunk 7]
```diff
@@ -146,19 +146,19 @@ std::shared_ptr<cadmium::dynamic::modeling::coupled<TIME>> NETWORK = std::make_s
...
-  cadmium::dynamic::translate::make_IC<Sender_defs::dataOut, inp_1>("sender1","Network"),
-  cadmium::dynamic::translate::make_IC<outp_2, Sender_defs::ackIn>("Network","sender1"),
-  cadmium::dynamic::translate::make_IC<Receiver_defs::out, inp_2>("receiver1","Network"),
-  cadmium::dynamic::translate::make_IC<outp_1, Receiver_defs::in>("Network","receiver1")
+  cadmium::dynamic::translate::make_IC<sender_defs::data_out, inp_1>("sender1","Network"),
+  cadmium::dynamic::translate::make_IC<outp_2, sender_defs::acknowledgement_in>("Network","sender1"),
+  cadmium::dynamic::translate::make_IC<receiver_defs::out, inp_2>("receiver1","Network"),
+  cadmium::dynamic::translate::make_IC<outp_1, receiver_defs::in>("Network","receiver1")
 };
 std::shared_ptr<cadmium::dynamic::modeling::coupled<TIME>> ABPSimulator = std::make_shared<cadmium::dynamic::modeling::coupled<TIME>>(
```
Comment (gagandeep8030) [inline src/top_model/main.cpp:163]: Maximum length<=80 characters per line

File: src/top_model/main.cpp (hunk 3 above)
Comment (revanth-sridhar) [reply src/top_model/main.cpp:131]: **We did not choose this file for coding conventions**. Please check below files, 

Files assigned to Manoj are:
- message.cpp (src/data_structures),
- main.cpp (test/src/sender),
- sendercadmium.hpp (include/atomics).

Files assigned to Revanth are:
- receivercadmium.hpp (include/atomics),
- subnetcadmium.hpp (include/atomics),
- message.hpp (include/data_structures)

File: src/top_model/main.cpp (hunk 4 above)
Comment (revanth-sridhar) [reply src/top_model/main.cpp:155]: **We did not choose this file for coding conventions**. Please check below files, 

Files assigned to Manoj are:
- message.cpp (src/data_structures),
- main.cpp (test/src/sender),
- sendercadmium.hpp (include/atomics).

Files assigned to Revanth are:
- receivercadmium.hpp (include/atomics),
- subnetcadmium.hpp (include/atomics),
- message.hpp (include/data_structures)

File: src/top_model/main.cpp (hunk 5 above)
Comment (revanth-sridhar) [reply src/top_model/main.cpp:134]: **We did not choose this file for coding conventions**. Please check below files, 

Files assigned to Manoj are:
- message.cpp (src/data_structures),
- main.cpp (test/src/sender),
- sendercadmium.hpp (include/atomics).

Files assigned to Revanth are:
- receivercadmium.hpp (include/atomics),
- subnetcadmium.hpp (include/atomics),
- message.hpp (include/data_structures)

File: src/top_model/main.cpp (hunk 6 above)
Comment (revanth-sridhar) [reply src/top_model/main.cpp:130]: **We did not choose this file for coding conventions**. Please check below files, 

Files assigned to Manoj are:
- message.cpp (src/data_structures),
- main.cpp (test/src/sender),
- sendercadmium.hpp (include/atomics).

Files assigned to Revanth are:
- receivercadmium.hpp (include/atomics),
- subnetcadmium.hpp (include/atomics),
- message.hpp (include/data_structures)

File: src/top_model/main.cpp (hunk 7 above)
Comment (revanth-sridhar) [reply src/top_model/main.cpp:163]: **We did not choose this file for coding conventions**. Please check below files, 

Files assigned to Manoj are:
- message.cpp (src/data_structures),
- main.cpp (test/src/sender),
- sendercadmium.hpp (include/atomics).

Files assigned to Revanth are:
- receivercadmium.hpp (include/atomics),
- subnetcadmium.hpp (include/atomics),
- message.hpp (include/data_structures)

File: include/atomics/senderCadmium.hpp [hunk 8]
```diff
@@ -27,148 +27,178 @@ using namespace cadmium;
...
-                if (state.ack){
-                  out.value = state.alt_bit;
-                  get_messages<typename defs::ackReceivedOut>(bags).push_back(out);
+            for(const auto &x : get_messages<typename defs::acknowledgement_in>(mbs)) {
+                if(state.model_active == true) {
+                    if (state.alt_bit == static_cast<int>(x.value)) {
+                        state.ack = true;
+                        state.sending = false;
+                        state.next_internal = TIME("00:00:00");
+                    }
```
Comment (gagandeep8030) [inline include/atomics/senderCadmium.hpp:148]: According to coding conventions, in nested loops, open and close brackets should be in the same column.

File: src/top_model/main.cpp (hunk 6 above)
Comment (gagandeep8030) [reply src/top_model/main.cpp:130]: Sorry for the confusion.

File: src/top_model/main.cpp (hunk 3 above)
Comment (gagandeep8030) [reply src/top_model/main.cpp:131]: sorry for the confusion.

File: src/top_model/main.cpp (hunk 5 above)
Comment (gagandeep8030) [reply src/top_model/main.cpp:134]: Sorry for the confusion.

File: src/top_model/main.cpp (hunk 4 above)
Comment (gagandeep8030) [reply src/top_model/main.cpp:155]: Sorry for the confusion.

File: src/top_model/main.cpp (hunk 7 above)
Comment (gagandeep8030) [reply src/top_model/main.cpp:163]: Sorry for the confusion.

File: test/src/sender/main.cpp [hunk 9]
```diff
@@ -13,129 +13,181 @@
...
+        cadmium::dynamic::translate::make_IC<
+            iestream_input_defs<message_t>::out, sender_defs::control_in
+        >("generator_con", "sender1"),
+        cadmium::dynamic::translate::make_IC<
+            iestream_input_defs<message_t>::out, sender_defs::acknowledgement_in
+        >("generator_ack", "sender1")
+    };
+    std::shared_ptr<cadmium::dynamic::modeling::coupled<TIME>>TOP 
+        = std::make_shared<cadmium::dynamic::modeling::coupled<TIME>>(
+          "TOP", submodels_TOP, iports_TOP, oports_TOP, eics_TOP,
```
Comment (gagandeep8030) [inline test/src/sender/main.cpp]: The variable names ending with "TOP" should be having its ending  with small letters.
For example  iports_TOP=> in_ports_top

File: include/atomics/senderCadmium.hpp (hunk 8 above)
Comment (manoj-goli) [reply include/atomics/senderCadmium.hpp:148]: As we discussed on the last Tuesday, these lines are code has no problem. Therefore, please approve the request.

**Comment by manoj-goli** [comment]
Hi, 

We are waiting for the approval of this pull request. The one comment that been posted has been resolved a couple of days back. Therefore, Kindly review and approve it.

**Review by manoj-goli — APPROVED** [review]

**Comment by gagandeep8030** [comment]
Hi Manoj,

Have you implemented the changes I mentioned for "TOP" in small letters? As I can still see its status as pending.

File: include/atomics/receiverCadmium.hpp [hunk 10]
```diff
@@ -29,84 +28,88 @@ using namespace cadmium;
...
-              }    
-              return next_internal;
-            }
+            }    
+            return next_internal;
+        }
 
-            friend std::ostringstream& operator<<(std::ostringstream& os, const typename Receiver<TIME>::state_type& i) {
-                os << "ackNum: " << i.ackNum; 
+        friend std::ostringstream& operator<<(std::ostringstream& os,
```
Comment (shubhamthakur463) [inline include/atomics/receiverCadmium.hpp]: As discussed earlier by Gagandeep, the variable names have to be self-descriptive. You can change "os" to something like "output_stream".
Thanks.

File: include/data_structures/message.hpp [hunk 11]
```diff
@@ -8,24 +8,19 @@
...
-  Message_t(float i_value)
-   :value(i_value){}
-
-  	float   value;
-
-  void clear() {
-   Message_t(); 
-  }
+struct message_t {
+	float value;
```
Comment (shubhamthakur463) [inline include/data_structures/message.hpp]: Kindly review this portion for indentation(4 Spaces).

File: include/atomics/subnetCadmium.hpp [hunk 12]
```diff
@@ -28,86 +28,96 @@ using namespace cadmium;
...
             }
+            else {
+                next_internal = std::numeric_limits<TIME>::infinity();
+            }    
+            return next_internal;
+        }
 
-            friend std::ostringstream& operator<<(std::ostringstream& os, const typename Subnet<TIME>::state_type& i) {
-                os << "index: " << i.index << " & transmiting: " << i.transmiting; 
+        friend std::ostringstream& operator<<(std::ostringstream& os,
```
Comment (shubhamthakur463) [inline include/atomics/subnetCadmium.hpp]: Again, make sure that variable names are self-explanatory.

File: include/data_structures/message.hpp (hunk 11 above)
Comment (revanth-sridhar) [reply include/data_structures/message.hpp]: Many thanks for letting us know on the indentation. We have fixed it now. May I request you to please review them?

File: include/atomics/subnetCadmium.hpp (hunk 12 above)
Comment (revanth-sridhar) [reply include/atomics/subnetCadmium.hpp]: Many thanks for letting us know about the variable names. We have fixed it now. May I request you to please review them ?

File: include/atomics/receiverCadmium.hpp (hunk 10 above)
Comment (revanth-sridhar) [reply include/atomics/receiverCadmium.hpp]: We have changed the variable names in the file. many thanks for letting us know. 
May I request you to please review them again?

File: test/src/sender/main.cpp (hunk 9 above)
Comment (manoj-goli) [reply test/src/sender/main.cpp]: Hi,
Name changes were made as per your comments in the 'respective files that we chose to modify for part B'. Please verify the code and kindly approve it.
Thanks!

**Review by manoj-goli — APPROVED** [review]

**Review by shubhamthakur463 — APPROVED** [review]
This pull has been APPROVED.
Thanks.

**Review by gagandeep8030 — APPROVED** [review]
Hi all I am approving the part B pull request

**Pull request merged**